opslevel/datasource_opslevel_team.go

```diff
@@ -1,85 +1,147 @@
+	},
+}
+
+func membershipsToListValue(ctx context.Context, members *opslevel.TeamMembershipConnection) basetypes.ListValue {
+	if members == nil {
+		return basetypes.NewListNull(memberObjectType)
+	}
+
+	output := make([]attr.Value, len(members.Nodes))
+	for i, membership := range members.Nodes {
+		object := make(map[string]attr.Value)
+		object["email"] = basetypes.NewStringValue(membership.User.Email)
+		object["role"] = basetypes.NewStringValue(membership.Role)
+		output[i] = types.ObjectValueMust(memberObjectType.AttrTypes, object)
+	}
+	return types.ListValueMust(memberObjectType, output)
+}
+
+func NewTeamDataSourceModel(ctx context.Context, team opslevel.Team) TeamDataSourceModel {
+	return TeamDataSourceModel{
+		Alias:       types.StringValue(team.Alias),
+		Id:          types.StringValue(string(team.Id)),
+		Members:     membershipsToListValue(ctx, team.Memberships),
+		Name:        types.StringValue(team.Name),
+		ParentAlias: types.StringValue(team.ParentTeam.Alias),
+		ParentId:    types.StringValue(string(team.ParentTeam.Id)),
+	}
+}
+
+func (teamDataSource *TeamDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_team"
+}
+
+func (teamDataSource *TeamDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "Team data source",
+
+		Attributes: map[string]schema.Attribute{
+			"alias": schema.StringAttribute{
+				MarkdownDescription: "The alias attached to the Team.",
+				Computed:            true,
+				Optional:            true,
+			},
+			"id": schema.StringAttribute{
+				Description: "The ID of this Team.",
+				Computed:    true,
+				Optional:    true,
+			},
+			"name": schema.StringAttribute{
+				Description: "The name of the Team.",
+				Computed:    true,
+			},
+			"members": schema.ListAttribute{
+				Description: "List of team members on the team with email address and role.",
+				ElementType: memberObjectType,
+				Computed:    true,
+			},
+			"parent_alias": schema.StringAttribute{
+				Description: "The alias of the parent team.",
+				Computed:    true,
+			},
+			"parent_id": schema.StringAttribute{
+				Description: "The id of the parent team.",
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func (teamDataSource *TeamDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var data TeamDataSourceModel
+
+	// Read Terraform configuration data into the model
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var err error
+	var team *opslevel.Team
+	if data.Alias.ValueString() != "" {
+		team, err = teamDataSource.client.GetTeamWithAlias(data.Alias.ValueString())
+	} else if data.Id.ValueString() != "" {
```

Comment:
```suggestion
	} else if !opslevel.IsID(data.Id.ValueString()) {
```
Alternatively could go first in the if else chain if `if opslevel.IsID(data.Id.ValueString()) {` is easier to read